webapp/src/main/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptor.java

```diff
@@ -77,7 +77,9 @@ public void afterCompletion(HttpServletRequest arg0,
 	 */
 	@Override
 	public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler, ModelAndView modelAndView) throws Exception {
-		modelAndView.addObject(SITE_PROPERTIES_ATTRIBNAME, this.siteProperties);
+		if (modelAndView!=null){
```

Comment:
I don't really understand what this does, but presumably there are some conditions under which `modelAndView` is null, and that is OK/expected?

Reply:
This postHandle appends configurable site-wide properties to the model so they are accessible to each view. If the request is returning something other than modelAndView it can't do this. My download test (which returns a downloaded key file rather than a modelAndView) was triggering a NPE. I don't know what else might trigger a NPE here, but thought it was safe to assume that if there is no modelAndView, then you can't append the properties to the model even if you wanted to.